src/wallet/rpcwallet.cpp

```diff
@@ -2008,6 +2016,9 @@ Value encryptwallet(const Array& params, bool fHelp)
 
     if (fHelp)
         return true;
+    if (!fEnableWalletEncryption) {
+        return false;
```

Comment:
I would suggest ``throw JSONRPCError(RPC_WALLET_ENCRYPTION_FAILED, "Error: Wallet encryption is disabled.");``